### ISSUE TITLE
Optimize1qGatesDecompositions does not optimize single (non-parameterized) epsilon rotations

### ISSUE THREAD

**Issue by kdk** [Qiskit/qiskit-terra#6473]
### Information

- **Qiskit Terra version**: main @ aceac9ea
- **Python version**: 3.6
- **Operating system**: osx 10.15.7

### What is the current behavior?

While looking into #6468 , `Optimize1qGatesDecompositions` will remove runs of (non-parameterized) single qubit gates if their net operation is sufficiently close to the identity, but will not do so for runs of only a single gate.

### Steps to reproduce the problem

```
>>> import qiskit as qk
>>> qc = qk.QuantumCircuit(1)
>>> qc.u1(0.0, 0)
>>> qc.draw()
     ┌───────┐
q_0: ┤ U1(0) ├
     └───────┘
>>> from qiskit.transpiler.passes import Optimize1qGates, Optimize1qGatesDecomposition
>>> Optimize1qGates(basis=['u1', 'u2', 'u3'])(qc).draw()

q_0:

>>> Optimize1qGatesDecomposition(basis=['u1', 'u2', 'u3'])(qc).draw()

q_0:

>>> qc = qk.QuantumCircuit(1)
>>> qc.u1(1e-17, 0)
>>> qc.draw()
     ┌───────┐
q_0: ┤ U1(0) ├
     └───────┘
>>> Optimize1qGates(basis=['u1', 'u2', 'u3'])(qc).draw()

q_0:

>>> Optimize1qGatesDecomposition(basis=['u1', 'u2', 'u3'])(qc).draw()
     ┌───────┐
q_0: ┤ U1(0) ├
     └───────┘
>>> qc = qk.QuantumCircuit(1)
>>> qc.u1(1e-17, 0)
>>> qc.u1(1e-17, 0)
>>> qc.draw()
     ┌───────┐┌───────┐
q_0: ┤ U1(0) ├┤ U1(0) ├
     └───────┘└───────┘
>>> Optimize1qGates(basis=['u1', 'u2', 'u3'])(qc).draw()

q_0:

>>> Optimize1qGatesDecomposition(basis=['u1', 'u2', 'u3'])(qc).draw()

q_0:

```

This may prevent the transpiler from making subsequent optimizations (without KAK) like
```
>>> qc = qk.QuantumCircuit(2)
>>> qc.cx(0, 1)
>>> qc.u1(1e-17, 0)
>>> qc.cx(0, 1)
>>> qc.draw()
          ┌───────┐
q_0: ──■──┤ U1(0) ├──■──
     ┌─┴─┐└───────┘┌─┴─┐
q_1: ┤ X ├─────────┤ X ├
     └───┘         └───┘
>>> Optimize1qGates(basis=['u1', 'u2', 'u3'])(qc).draw()

q_0: ──■────■──
     ┌─┴─┐┌─┴─┐
q_1: ┤ X ├┤ X ├
     └───┘└───┘
>>> Optimize1qGatesDecomposition(basis=['u1', 'u2', 'u3'])(qc).draw()
          ┌───────┐
q_0: ──■──┤ U1(0) ├──■──
     ┌─┴─┐└───────┘┌─┴─┐
q_1: ┤ X ├─────────┤ X ├
     └───┘         └───┘
```

**Comment by mtreinish** [comment]
We probably just need to add some tolerance to this line: https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/transpiler/passes/optimization/optimize_1q_decomposition.py#L83 basically just swap that with numpy `equiv` instead of `equals` (using approximation degree for the tolerance might make sense too)

**Comment by mwessling90** [comment]
Hi, i am new and would like to work on this if thats ok.

**Comment by mtreinish** [comment]
@mwessling90 sure, I've assigned the issue to you. Feel free to reach out if you need any help

**Comment by fs1132429** [comment]
I want to work on this issue if thats ok.

**Comment by javabster** [comment]
Thanks @fs1132429 but this issue has already been assigned to someone else, I saw you volunteered for a few other issues so I've assigned a couple others to you 😄

**Comment by mattwright99** [comment]
@javabster it looks like @mwessling90 might be MIA. Can I take over this issue?

**Comment by javabster** [comment]
> @javabster @mwessling90 would it be ok if I took over this issue?

Hi @mattwright99 sure I'll reassign to you as @mwessling90 is unresponsive 😄

**Comment by levbishop** [comment]
Is this fixed by #6553 ?

**Comment by mattwright99** [comment]
@levbishop I checked the logic and ran the tests again - same result as before. I will resolve the conflicts and update my PR so that this issue can be closed.
